tt_metal/llrt/tt_cluster.cpp

```diff
@@ -37,6 +37,8 @@ Cluster::Cluster() {
 
     this->detect_arch_and_target();
 
+    this->initialize_hal();
```

Comment:
It makes sense to initialize the HAL in here but can you remove this call from `impl/device.cpp`

Reply:
Addressed in commit https://github.com/tenstorrent/tt-metal/pull/13510/commits/338a6322328b38d5b46c4210eeabd781a6c10e53